### ISSUE TITLE
Algolia Crawler using Nuxt

### ISSUE THREAD

**Issue by yassilah** [nuxt-modules/algolia#8]
### Is your feature request related to a problem? Please describe.
No.

### Describe the solution you'd like
The idea would be to add the ability to store pages' metadata into an Algolia index during Nuxt generation process. This would then allow for instant search through all generated pages content.

### Describe alternatives you've considered
The existing [Algolia Crawler](https://www.algolia.com/doc/tools/crawler/getting-started/overview/) does basically the same thing but having it as a Nuxt module has several advantages:
- the Crawler is available but as a [paid add-on](https://www.algolia.com/pricing/);
- there is a [Netlify plugin](https://www.algolia.com/doc/tools/crawler/netlify-plugin/quick-start/) but having the process done during Nuxt generation would allow for a host agnostic content indexing;
- it would give full control over how and which data is indexed during the generation process using Nuxt hooks;
- it should be quite straightforward to use the `useSearch` composable and even create a simple `<NuxtPagesSearch />` component (or whatever we might want to call it).

### Additional context
For the crawler to work we would need some extra module options. Here's a suggestion of how it could look like:

```ts 
interface Options {
    applicationId: string;
    indexName: string;
    search: {
        apiKey: string
        apiAdminKey: string
        fields?:
            | ((route: string, html: string) => Record<string, string>)
            | string[] // array of metadata fields to include in the index or function that returns a metadata object 
        include?: ((route: string) => boolean) | (string | RegExp)[] // array of routes to include or function to filter out routes
    }
}

**Comment by Baroshem** [comment]
As soon as the guys from vue-instantsearch merge my recommendations to the code, we may be able to use vue 3 search components as a part of this module.

This is really an amazing piece of work! After some small additions I will gladly merge it and release with the new version :)

**Comment by Baroshem** [comment]
Closed with #9
